File: resources/lua/modules/webots/wbfile.lua
local M = {}

-- remove the filename and extension from the file path
function M.getpathwithoutfilename(filePath)
  -- make sure to use '/' as separator
  filePath = string.gsub(filePath, "\\", "/")
  -- reverse the string in order to find the 'first' separator
  filePath = string.reverse(filePath)
  local index = string.find(filePath, "/")
  -- remove everything that is 'before the first' separator and re-reverse the string
  filePath = string.sub(filePath, index + 1)
  filePath = string.reverse(filePath)
  return filePath
end

return M

File: resources/javascript/modules/webots/wbfile.js
/*
 * content: utility functions to handle files
 */

export function testFunction() { // TODO: to remove
  return 'WBFILE WORKS';
};

// remove the filename and extension from the file path
export function getPathWithoutFilename(filePath) {
  // make sure to use '/' as separator
  filePath = filePath.replace('\\', '/');
  const r = /[^\/]*$/;

  return filePath.replace(r, '');
}
